### PR TITLE
Added nesting of sidebar items

### PR DESCRIPTION
**Pull request by Om-Doiphode**
Fixes issue #713 

- Changed the documentation theme from `furo` to `pydata-sphinx-theme`
-  Implemented nesting of menus and submenus in the `Section Navigation`.
- Improved the overall structure and navigation of the documentation.
- Ensured that all relevant sections are properly organized under their respective categories.

image
image
image
image

### CODE REVIEW

**Review by henrykironde — CHANGES_REQUESTED** [review]
@Om-Doiphode, thank you for these changes. The documentation updates align well with what we discussed. However, I have a few suggestions:

It's standard practice to keep the `source` or `api` directories dedicated solely to API module documentation. This helps streamline maintenance when the source code is updated. The `conf.py` file should remain in the default location: `docs/conf.py`. I don't think we need the `deepforest_docs` folder. We can remove it, and all the subdirectories can be moved directly under `docs/`.

We still have some redundant  parts like getting started appear twice, in the[ introduction]( https://deepforest--752.org.readthedocs.build/en/752/deepforest_docs/introduction/getting_started.html#) and again as a [main topic](https://deepforest--752.org.readthedocs.build/en/752/deepforest_docs/getting_started/index.html).

**Comment by henrykironde** [comment]
Can you rebase with main

**Comment by henrykironde** [comment]
Make this one commit. Amend the commits to one

**Comment by bw4sz** [comment]
@Om-Doiphode Once you rebase, i'm excited to get this merged. Let us know if you need help.

File: dev_requirements.txt [hunk 1]
```diff
@@ -4,7 +4,8 @@ aiohttp
 bumpversion
 comet_ml
 docformatter
-furo
+docutils<0.18
```
Comment (henrykironde) [inline dev_requirements.txt:7]: remove this dep, "docutils<0.18"

**Review by henrykironde — CHANGES_REQUESTED** [review]
Make one commit. 
Make sure the tests pass. Look at the error message and see if you can figure out the reason why it is failing.

**Comment by henrykironde** [comment]
@Om-Doiphode Use `pytest --nbmake docs/**/*_test.ipynb` to test specific notebooks. To exclude certain notebooks from CI/CD runs, ensure the ones you want to test end with `_test.ipynb`. This approach avoids running time-consuming notebooks that should be tested locally. Make sure when you run  `ls docs/**/*_test.ipynb` locally, you retriever the files expected to be tested.

**Comment by bw4sz** [comment]
@Om-Doiphode, rebase, there was another PR.

**Pull request closed without merging**